src/lib.rs

```diff
@@ -3,6 +3,7 @@
 mod cipher;
 pub mod config;
 pub mod context;
+mod error;
```

Comment:
should mod error be public so that error can be propagated to its user for correct handling?

Reply:
Done! 😄